ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -50,15 +50,7 @@ const FlowCanvas: React.FC<Props> = () => {
   const onInit = async (reactFlowInstance) => {
     setReactFlowInstance(reactFlowInstance);
 
-    let store = getDataStore();
-
-    /**
-     * TODO: we should have a promise that resolves when the datastore is loaded,
-     * becuase this might cause problems down the road when it's in both Home.tsx
-     * and here.
-     */
-    await store.fetchOrCreateRoot();
-
+    let store = await getDataStore();
```

Comment:
I suppose the other way of doing this would be to keep the same style of `getDataStore()`, but also have a promise somewhere, like `store.loaded`. However, I like the idea of piping the store in as a prop. Any changes to it could also call a delegate that calls the `setStore` hook and causes a re-render.